sdk/search/search-documents/test/internal/serialization.spec.ts

```diff
@@ -16,7 +16,29 @@ describe("serialization.serialize", () => {
   it("circular", () => {
     const circluarInput: any = { a: null };
     circluarInput.a = circluarInput;
-    assert.throws(() => serialize(circluarInput));
+    const result = serialize(circluarInput);
+    assert.deepEqual(circluarInput, result);
+  });
+
+  it("recursive 1", () => {
+    const child = { hello: "world" };
+    const documents = [
+      { id: "1", children: [child] },
+      { id: "2", children: [child] }
+    ];
+    const result = serialize(documents);
+    assert.deepEqual(documents, result);
+  });
+
+  it("recursive 2", () => {
+    const child = { hello: Infinity, world: -Infinity, universe: NaN };
+    const documents = [
+      { id: "1", children: [child] },
+      { id: "2", children: [child] },
+      { id: "3", children: [child] }
+    ];
+    const result = serialize(documents);
+    assert.deepEqual(documents, result);
```

Comment:
wouldn't this fail because the result shape is `{hello: 'INF', world: '-INF', universe: 'NaN'}` instead of the number value after serialization?